packages/@aws-cdk/aws-iot-actions/lib/kinesis-put-record-action.ts

```diff
@@ -0,0 +1,54 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import * as kinesis from '@aws-cdk/aws-kinesis';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Configuration properties of an action for the Kinesis Data stream.
+ */
+export interface KinesisPutRecordActionProps extends CommonActionProps {
+  /**
+   * The partition key used to determine to which shard the data is written.
+   * The partition key is usually composed of an expression (for example, ${topic()} or ${timestamp()}).
+   * For more information @see https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecord.html#API_PutRecord_RequestParameters
+   *
+   * @default - None
+   */
+  readonly partitionKey?: string;
+}
+
+/**
+ * The action to put the record from an MQTT message to the Kinesis Data stream.
+ */
+export class KinesisPutRecordAction implements iot.IAction {
```

Comment:
I don't like the fact that this name is inconsistent with the name for the Firehose Action, which is `FirehoseStreamAction.

So, I think we need to do one of two things:

1. Either Rename `FirehoseStreamAction` to `FirehosePutRecordAction`, or:
2. Rename this class from `KinesisPutRecordAction` to `KinesisStreamAction`.

I'll let you decide which one you prefer 🙂.

Reply:
I think `1` is better! I will create the PR for rename😉 .

Reply:
https://github.com/aws/aws-cdk/pull/18356